flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/KeyedStream.scala

```diff
@@ -183,7 +183,10 @@ class KeyedStream[T, K](javaStream: KeyedJavaStream[T, K]) extends DataStream[T]
    * Creates a new [[DataStream]] by folding the elements of this DataStream
    * using an associative fold function and an initial value. An independent 
    * aggregate is kept per key.
+   *
+   * @deprecated will be removed in a future version
    */
+  @Deprecated
```

Comment:
Scala has `@deprecated` where you can also specify the reason right in the annotation. (https://coderwall.com/p/k4ulyg/scala-deprecated-annotation)

Also holds for the other instances in Scala code.